pandas/core/internals.py

```diff
@@ -1800,6 +1820,91 @@ def _unstack(self, unstacker_func, new_columns):
         return blocks, mask
 
 
+class ExtensionBlock(NonConsolidatableMixIn, Block):
+    """Block for holding extension types.
+
+    Notes
+    -----
+    This holds all 3rd-party extension array types. It's also the immediate
+    parent class for our internal extension types' blocks, CategoricalBlock.
+
+    ExtensionArrays are limited to 1-D.
+    """
+    def __init__(self, values, placement, ndim=None):
+        self._holder = type(values)
```

Comment:
I think _holder can/should be a property of the Block itself (it can be a cached property)

Reply:
`Block` defines `_holder` as a class attribute. Why would caching it be helpful? It's not reused among multiple `ExtensionBlocks`.

Reply:
its just 1 more thing to do in the __init__ which is not needed